src/system/newlib_stubs.c

```diff
@@ -56,3 +60,40 @@ int getentropy(void *_buffer, size_t _length) {
 void __sync_synchronize(void) {
 	__sync_synchronize();
 }
+
+int clock_gettime(clockid_t clock_id, struct timespec* tp) {
+	struct timeval tv;
+	int retval = -1;
+
+	switch (clock_id)
+	{
+	case CLOCK_REALTIME:
+		retval = gettimeofday(&tv, NULL);
+		if (!retval) TIMEVAL_TO_TIMESPEC(&tv, tp);
+		break;
+	
+	default:
+		errno = EINVAL;
+		break;
+	}
+
+	return retval;
+}
+
+
+// The value for this variable is added by the common.mk makefile so that the
+// timestamp is up to date with the hot package's latest compilation
+extern char const* _PROS_COMPILE_TIMESTAMP_INT;
+
+int _gettimeofday(struct timeval* tp, void* tzvp) {
+	if (competition_is_connected()) {
+		tp->tv_sec = vexSystemTimeGet() * 1000;
+		tp->tv_usec = vexSystemHighResTimeGet();
```

Comment:
Is this what you expect when tested? The docs say this derives from a hardware timer started on startup. You're also deriving this information from 2 vex syscalls, so getting the times aren't atomic.

Reply:
honestly the times in its current form are probably atomic *enough* given he's getting the seconds and then the microseconds.